api/provider/aws/load_balancer_delete.go

```diff
@@ -44,6 +47,72 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 		}
 	}
 
+	getRoleFN := func() error {
+		// Check IAM Role
+		input := &iam.GetRoleInput{}
+		input.SetRoleName(roleName)
+
+		if _, err := l.AWS.IAM.GetRole(input); err != nil {
+			if err, ok := err.(awserr.Error); ok && err.Code() == "NoSuchEntity" {
+				return nil
+			}
+			return err
+		}
+		return nil
+	}
+
+	getRolePolicyeFN := func() error {
+		// Check Role Policy
+		input := &iam.GetRolePolicyInput{}
+		input.SetRoleName(roleName)
+		input.SetPolicyName(policyName)
+
+		if _, err := l.AWS.IAM.GetRolePolicy(input); err != nil {
+			if err, ok := err.(awserr.Error); ok && err.Code() == "NoSuchEntity" {
+				return nil
+			}
+			return err
+		}
+		return nil
+	}
+
+	readSecurityGroupFN := func() error {
+		// Check Security Group
+		if _, err := readSG(l.AWS.EC2, securityGroupName); err != nil && !strings.Contains(err.Error(), "does not exist") {
+			return err
+		}
+		return nil
+	}
+
+	desribeLoadBalancerFN := func() error {
+		if _, err = describeLoadBalancer(l.AWS.ELB, fqLoadBalancerID); err != nil {
+			if serverError, ok := err.(*errors.ServerError); ok && serverError.Code == "LoadBalancerDoesNotExist" {
+				return nil
+			}
+			return err
+		}
+		return nil
+	}
+
+	ch := make(chan error, 4)
+	defer close(ch)
+
+	go retry(10*time.Second, 2*time.Second, ch, getRoleFN)
```

Comment:
Would love advice for cleaning this section up. Is it possible to do an array of functions? Or some helper function.